### ISSUE TITLE
Root Dataverse (Harvard) appears in sword service document on dataverse-demo

### ISSUE THREAD

**Issue by rliebz** [IQSS/dataverse#1443]
On a new Dataverse user:
`curl -u $USERNAME:$PASSWORD https://dataverse-demo.iq.harvard.edu/dvn/api/data-deposit/v1.1/swordv2/service-document` includes the root Dataverse, while the same request using the host `apitest.dataverse.org` does not include the root Dataverse. 

```xml
<service xmlns="http://www.w3.org/2007/app" xmlns:atom="http://www.w3.org/2005/Atom”>
  <workspace>
    <atom:title type="text">Harvard</atom:title>
    <collection href="https://dataverse-demo.iq.harvard.edu/dvn/api/data-deposit/v1.1/swordv2/collection/dataverse/root”>
      <atom:title type="text">Harvard</atom:title>
```

I do not expect that the root Dataverse should appear in either request.

**Comment by pdurbin** [comment]
In #1070 I noted that the Service Document checks the AddDataset but probably other SWORD operations are using a stricter check (the permission names have changed but currently it's EditDataverse). So there's a mismatch here and I was hoping to move more full to the new permission system in that ticket. Once we resolve this (which will be some work) I don't see any reason why we can't allow deposit into the root for a Dataverse installation that has been configured for it.

(In 3.x there was simply a check for "admin" that we required for all SWORD operations.)

Also, as @scolapasta mentioned, we should look at how apitest and demo are configured in terms of what authenticated users are allowed to do. I give the "dvContributor" role on apitest (with [some scripts](https://github.com/IQSS/dataverse/blob/master/scripts/deploy/apitest.dataverse.org/post)) but I need to figure out how demo is set up. Perhaps "dsContributor" is also given.

**Comment by scolapasta** [comment]
As Phil noted, the issue is really what is in #1070 which is that the Service Document is returning what should be correct (does the user have the Add Dataset permission), but the Sword Create Dataset command is not using that permission to check.

**Comment by pdurbin** [comment]
@scolapasta a temporary work around might be to change it so the Service Document check **doesn't** use the fancy new permissions system.

**Comment by scolapasta** [comment]
Closing this issue as the behavior is correct and what needs to be fixed is #1070.
